controllers/pvc-mutator/encryption/keygen.go

```diff
@@ -101,21 +101,29 @@ func (s *EncryptionKeySetter) MutatePVC(ctx context.Context, pvc *corev1.Persist
 	log := s.log.WithValues("pvc", client.ObjectKeyFromObject(pvc).String())
 	log.V(4).Info("received pvc for mutation")
 
-	// Skip mutation if the PVC is not provisioned by StorageOS
-	provisioned, err := provisioner.IsProvisionedPVC(s.Client, *pvc, namespace, provisioner.DriverName)
+	// Retrieve StorageClass of the PVC.
+	storageClass, err := provisioner.StorageClassForPVC(s.Client, pvc)
 	if err != nil {
-		return errors.Wrap(err, "failed to check pvc provisioner")
+		return errors.Wrap(err, "failed to retrieve storageclass of pvc")
 	}
+
+	// Skip mutation if the PVC is not provisioned by StorageOS
+	provisioned := provisioner.IsProvisionedStorageClass(storageClass, provisioner.DriverName)
 	if !provisioned {
 		log.V(4).Info("pvc does not provisioned by StorageOS, skipping")
```

Comment:
Can you change this while you're here:

`pvc does not provisioned by StorageOS, skipping` -> `pvc will not be provisioned by StorageOS, skipping`